model-optimizer/extensions/middle/UnsqueezeTileReshapeBlockToInterpolate.py

```diff
@@ -76,79 +80,110 @@ def pattern(self):
             ]
         )
 
-    def replace_pattern(self, graph: Graph, match: dict):
+    @staticmethod
+    def is_applicable(match: dict) -> bool:
+        """
+        This function checks whether this transformation is applicable.
+        :param match: dictionary with nodes from the found pattern
+        :return: True, if the transformation is applicable
+                 False, otherwise
+        """
         unsqueeze_node = match['unsqueeze']
-        unsqueeze_name = unsqueeze_node.name
-
         second_input_of_unsqueeze = unsqueeze_node.in_port(1).get_connection().get_source().node
-        if not second_input_of_unsqueeze.has_valid('value'):
-            return
+        if not second_input_of_unsqueeze.has_valid('value') or len(second_input_of_unsqueeze.value) != 1:
+            return False
 
         d_idx = int(second_input_of_unsqueeze.value)
 
         second_input_of_tile = match['tile'].in_port(1).get_connection().get_source().node
         if not second_input_of_tile.has_valid('value'):
-            return
+            return False
 
         input_shape_of_unsqueeze = unsqueeze_node.in_port(0).data.get_shape()
-        if len(input_shape_of_unsqueeze) not in {4, 5}:
+        input_rank_of_unsqueeze = len(input_shape_of_unsqueeze)
+        if input_rank_of_unsqueeze not in {4, 5}:
+            return False
+
+        if input_rank_of_unsqueeze + 1 != len(second_input_of_tile.value):
+            return False
+
+        expected_tile_constant = np.ones(input_rank_of_unsqueeze + 1)
+        expected_tile_constant[d_idx] = float(second_input_of_tile.value[d_idx])
+
+        if not np.array_equal(expected_tile_constant, float32_array(second_input_of_tile.value)):
+            return False
+
+        reshape_node = match['reshape']
+        new_shape = reshape_node.in_port(1).data.get_value()
+        if new_shape is None or input_rank_of_unsqueeze != len(new_shape):
+            return False
+
+        return True
+
+    def replace_pattern(self, graph: Graph, match: dict):
+        if not self.is_applicable(match):
             return
 
-        scale = float32_array([second_input_of_tile.value[d_idx]])
+        unsqueeze_node = match['unsqueeze']
+        unsqueeze_name = unsqueeze_node.soft_get('name', unsqueeze_node.id)
+        second_input_of_unsqueeze = unsqueeze_node.in_port(1).get_connection().get_source().node
+        d_idx = int(second_input_of_unsqueeze.value)
         axis = d_idx - 1
```

Comment:
What would it be if `d_idx=0`?

Reply:
`Unsqueeze` inserts the dimension `1` **after** the interpolated axis. Hence, we should use `d_idx - 1`  as the interpolated axis.